oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/accesscontrol/ACE.java

```diff
@@ -112,7 +111,7 @@ public boolean isAllow() {
     @NotNull
     @Override
     public String[] getRestrictionNames() {
-        return Collections2.transform(restrictions, this::getJcrName).toArray(new String[restrictions.size()]);
+        return restrictions.stream().map(this::getJcrName).toArray(size -> new String[size]);
```

Comment:
I'd prefer `String[]::new` over `size -> new String[size]`.

Reply:
But that won't pre-allocate the array with the suggested size, right?